src/constants.js

```diff
@@ -46,15 +46,21 @@ function getDirectory(category: string): string {
   return path.join(userHome, `.${category}`, 'yarn');
 }
 
-function getCacheDirectory(): string {
+function getPreferredCacheDirectories(): Array<string> {
+  const preferredCacheDirectories = [getDirectory('cache')];
+
+  if (process.platform !== 'win32') {
+    preferredCacheDirectories.unshift('/tmp/.yarn-cache');
+  }
+
   if (process.platform === 'darwin') {
-    return path.join(userHome, 'Library', 'Caches', 'Yarn');
+    preferredCacheDirectories.unshift(path.join(userHome, 'Library', 'Caches', 'Yarn'));
```

Comment:
This kind of changes our previous logic so not sure about that. We should directly fallback to `tmp` when our first choice fails. With this one, it will try this on OS X and then try the Linux version second.